projectboard/main.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright © 2023 The Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
```

Comment:
Can we add a new pkg for `general`/`gh-projects` instead of main.go?

Reply:
If this repo is just going to be a collection of tools that can be run using `go run main.go`, then we don't need packages. IMO, It would be easier to have lightweight tools that perform a specific task instead of a single `main.go` that does a bunch of things?

Reply:
What will happen if other SIGs want to add their GitHub action or code to run GitHub actions similar to this PR? I recommend organizing it so more SIGs can contribute and reuse every piece as an independent feature. 

This was the intent from this repo, as I described in the issue https://github.com/kubernetes/community/issues/7316#issue-1726092953

Reply:
> What will happen if other SIGs want to add their GitHub action or code to run GitHub actions similar to this PR? I recommend organizing it so more SIGs can contribute and reuse every piece as an independent feature.

I'm not sure I understand. Are you saying in the future, SIGs will add `<sig-name>/gh-projects>` with similar code but specific to theirs? This code is generic and every SIG would have it's own file in `.github/workflows` reusing the same code? The `main.go` file is in a specific `projectboard` dir that can and _should_ (hopefully) reused for all the SIGs. Is the recommendation to rename the dir?

Reply:
For some reason, I didn't see the folder name .... nevermind :) 

This code is generic and should be the base for SIGs. However, what if one SIG needs to add a new filter or update this code? Shall we move the login into a sub pkg? so it will be more reusable?

Reply:
@helayoty @aramase we can take some ideas from knative/actions. I had a discussion with @MadhavJivrajani and kind of got the crux of what we are aiming to achieve . Knative/actions works in different manner. They basically have common templates that are leveraged by the community. Our approach would be to manage actions and run them from a central repo. From this issue I am inferring that this directory is basically going to act like a collection of hack to be used by the code owners and maintainers along with a bit more thing. The way we are aiming is definitely going to be different than knative/action so not sure whether we can get some good idea from this one, but maybe taking a look on that wouldn't hurt. Anyways, this script basically adds functionality to add the stuffs in your issue and prs to project board. A suggestion here would be to add the date range as well. Also, for PRs tree and blob-based filtering would be a good idea as different sigs have ownership of different codebase so this might be a useful feature.

I also had few questions. LIke as in Knative community, they have template action for example for formatting, for release, etc that are generic and they are directly leveraged by the community. So basically though each repo has it's own .github/workflow. But they are directly exporting it like this https://github.com/knative/operator/blob/main/.github/workflows/knative-stale.yaml and https://github.com/knative/actions/blob/main/.github/workflows/reusable-stale.yaml. Now my question is why do we want to maintain and run all the github actions from a monorepo. If maintenance is an issue, can't we just go with the knative approach and create multiple hack scipt like we are currently doing. I have no idea of your approach so my question might be a bit negligent. Kindly correct me If I am wrong somewhere and also let me know about your's approach and it's pros and cons over knative approach. Thanks!

Reply:
One more thing, in any case, can we start opening feature-based issues so that I can get some more details of the thought plan and provide some help as well?

Reply:
> This code is generic and should be the base for SIGs. However, what if one SIG needs to add a new filter or update this code? Shall we move the login into a sub pkg? so it will be more reusable?

+1 to @helayoty's comment.

Reply:
@sbdtu5498 the main intent of this is to consolidate actions work that's happening in the Kubernetes community.
The approach you suggested works well, but another aspect of consolidation that's desirable by doing things this way is security. There's some security concerns we've had with running GitHub actions, having everything in one place makes it uniform and easy for the GitHub admin team to take action.

Also @sbdtu5498, the feature requests you suggested, please feel free to create issues for them and contribute! Thanks a bunch for your feedback btw ❤️